### PR TITLE
A council can delete draft proposal

### PR DESCRIPTION
**Pull request by billie66**
Finished features:

* Add delete button on proposal detail page
* Ask for user confirmation before delete operation
* Format code

UI like this:

Screen Shot 2019-07-25 at 8 57 53 AM

### CODE REVIEW

File: back-end/src/service/CVoteService.ts [hunk 1]
```diff
@@ -100,6 +100,30 @@ export default class extends Base {
     }
   }
 
+  // delete draft proposal by proposal id
+  public async deleteDraft(param: any): Promise<any> {
+    const db_cvote = this.getDBModel('CVote')
+    if (!this.currentUser || !this.currentUser._id) {
+      throw 'cvoteservice.deleteDraft - invalid current user'
+    }
+    if (!this.canManageProposal()) {
```
Comment (luckyyang) [inline back-end/src/service/CVoteService.ts]: we dont need permission check here, you can add access rules for route level on admin panel /admin/permission

**Comment by luckyyang** [comment]
Please add a test for deleteDraft method in service code @billie66

File: back-end/src/service/CVoteService.ts (hunk 1 above)
Comment (billie66) [reply back-end/src/service/CVoteService.ts]: I copied some code from updateDraft method, but I don't know this rule, I will check it

**Comment by billie66** [comment]
ok, i will write test code for this method

**Comment by billie66** [comment]
ok, i will write test code for this method

**Comment by luckyyang** [comment]
you don't have to close the pr, just keep pushing commits to this branch, they will be added accordingly

**Pull request merged**